src/ripple/overlay/impl/OverlayImpl.h

```diff
@@ -456,6 +456,60 @@ class OverlayImpl : public Overlay
 
     void
     sendEndpoints();
+
+private:
+    struct Stats
+    {
+
+        template <class Handler>
+        Stats (Handler const& handler, beast::insight::Collector::ptr const& collector)
+            : hook (collector->make_hook (handler))
+            , peerDisconnects (collector->make_gauge("Overlay","Peer_Disconnects")) 
+            { 
+                mcollector=collector;
+            }
+
+        beast::insight::Hook hook;
+        beast::insight::Gauge peerDisconnects;
+        std::vector<beast::insight::Gauge> trafficGauges;
```

Comment:
I know you didn't choose the name of `beast::insight::Gauge`, but can we call this member `trafficCounters` instead of `trafficGauges`? Going by StatsD definitions, these are counters, not gauges. I'd love to see `beast::insight::Gauge` renamed to `Counter` as well, if it is in fact a counter and not a gauge, but I haven't looked.

Reply:
Since the value we are exporting is already an aggregate statistic I am exporting it as a gauge so that the metrics are not double counted. I understand that it is confusing to call this a gauge, but I will be sure to document the rationale for this in the monitoring tool.